apps/assisted-disconnected-ui/src/components/ResetSingleClusterModal.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import {
+  Button,
+  Modal,
+  ModalVariant,
+  ButtonVariant,
+  Text,
+  TextContent,
+} from '@patternfly/react-core';
+import { useModalDialogsContext, ClustersService } from '@openshift-assisted/ui-lib/ocm';
+import { getApiErrorMessage, handleApiError, ErrorState } from '@openshift-assisted/ui-lib/common';
+import { useNavigate } from 'react-router-dom-v5-compat';
+
+const ResetSingleClusterModal: React.FC = () => {
+  const navigate = useNavigate();
+  const [error, setError] = React.useState<{ title: string; message: string } | null>(null);
+  const { resetSingleClusterDialog } = useModalDialogsContext();
+  const { data, isOpen, close: onClose } = resetSingleClusterDialog;
+  const cluster = data?.cluster;
+
+  if (!cluster) {
+    return null;
+  }
+
+  const handleClose = () => {
+    setError(null);
+    onClose();
+  };
+
+  const handleResetAsync = async () => {
+    try {
+      await ClustersService.remove(cluster.id);
+      navigate(`/`);
+    } catch (e) {
+      handleApiError(e, () => {
+        setError({
+          title: 'Failed to reset cluster installation',
+          message: getApiErrorMessage(e),
+        });
+      });
+    }
+  };
```

Comment:
Please add i18n for the errror title.

We should also have loading state & reset the error (in case user retries)
```suggestion
    try {
      setError(undefined)
      setIsLoading(true)
      await ClustersService.remove(cluster.id);
      navigate(`/`);
    } catch (e) {
      handleApiError(e, () => {
        setError({
          title: 'Failed to reset cluster installation',
          message: getApiErrorMessage(e),
        });
      });
    } finally {
      setIsLoading(false);
    }
  };
```